daemon/archive.go

```diff
@@ -43,7 +44,17 @@ func archivePath(i interface{}, src string, opts *archive.TarOptions, root strin
 	if ap, ok := i.(archiver); ok {
 		return ap.ArchivePath(src, opts)
 	}
-	return chrootarchive.Tar(src, opts, root)
+
+	// handle a case where the source path that was passed in is actually the parent of `root`
+	// this would happen when the requested archive path *is* the root.
+	if filepath.Dir(root) == src {
+		root = src
```

Comment:
this is not good because it only works for overlay2. For vfs, since the layout is different (no `merged` directory), src will be equal to `/var/lib/docker/vfs/dir` while root will be `/var/lib/docker/vfs/dir/$ID`. You don't want to tar all of `/var/lib/docker/vfs/dir`.

Reply:
I debugged root and src when running `-e DOCKER_GRAPHDRIVER=vfs`

Reply:
This is the chroot path, not changing the source path.
I'm running my new test against vfs and it seems to be ok.

Reply:
And CI is running vfs as well.

Reply:
I edited the first comment of this thread, I incorrectly said it would tar all the vfs dirs, when in fact the problem is that the chroot would not happen at the correct directory.

Reply:
@cpuguy83 sorry i didn't see your comments here (got github cached). I wasn't expecting the tests to fail under vfs.